src/Domain/SurfaceElement.cpp

```diff
@@ -0,0 +1,103 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Domain/SurfaceElement.hpp"
+
+#include <cstddef>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/EagerMath/Determinant.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/Structure/Direction.hpp"
+#include "Evolution/DiscontinuousGalerkin/ProjectToBoundary.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/ContainerHelpers.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+
+template <size_t VolumeDim, typename TargetFrame>
+void euclidean_surface_element(
+    const gsl::not_null<Scalar<DataVector>*> result,
+    const InverseJacobian<DataVector, VolumeDim, Frame::ElementLogical,
+                          TargetFrame>& inverse_jacobian_face,
+    const Scalar<DataVector>& inverse_jacobian_determinant_face,
+    const Direction<VolumeDim>& direction) {
+  ASSERT(inverse_jacobian_face.get(0, 0).size() ==
```

Comment:
prefer `get<0, 0>(tensor)` when you know the indices at compile time. Same for `get(scalar)` over `scalar.get()`.